search-service/src/main/kotlin/com/egm/stellio/search/temporal/service/TemporalService.kt

```diff
@@ -64,14 +72,7 @@ class TemporalService(
     ): Either<APIException, Unit> = either {
         authorizationService.userCanCreateEntities(sub.toOption()).bind()
 
-        // create a view of the entity containing only the most recent instance of each attribute
-        val expandedEntity = ExpandedEntity(
-            sortedJsonLdInstances
-                .keepFirstInstances()
-                .addCoreMembers(jsonLdTemporalEntity.id, jsonLdTemporalEntity.types)
-        )
-        val ngsiLdEntity = expandedEntity.toNgsiLdEntity().bind()
-
+        val (expandedEntity, ngsiLdEntity) = parseExpandedInstances(sortedJsonLdInstances, jsonLdTemporalEntity).bind()
         entityService.createEntity(ngsiLdEntity, expandedEntity, sub).bind()
```

Comment:
Maybe you should have an already exist error here even if the entity is marked as deleted.

Reply:
what is done in `EntityService` when you try to create an entity that already exists
- if not marked as deleted: already exists error (as before)
- if marked as deleted: ok if you are admin of the entity, forbidden if not

does not make sense?

Reply:
I thought that we were not supposed to create a temporal entity if it already existed. But that is not the case. The post endpoint is specified as update Or Create.